CMakeLists.txt

```diff
@@ -235,6 +231,29 @@ if(CUDA_FOUND)
       message(WARNING "On some Unix systems CUDA 10.0+ requires CMake 3.12.2+; you use CMake ${CMAKE_VERSION}")
   endif()
 
+  # We want to compile as many targets as possible but different CUDA versions support different targets.
+  # Let's instead enable options based on what cuda version we have.
```

Comment:
No. I have specifically removed options not supported by the CUDA version. The user will only be able to set options supported by the CUDA version they are using. Otherwise the compilation will either fail, or produce warnings about the use of deprecated architecture.